### ISSUE TITLE
UI Won't update after deleting a goal

### ISSUE THREAD

**Issue by DylanJCain** [bradtraversy/mern-tutorial#5]
`(goal) => goal._id !== action.payload.id`

needs changed to:

`(goal) => goal._id !== action.payload._id`

**Comment by ISorokos** [comment]
Not clear why this is the case for you, but keeping it as is works for me.
What you're proposing is breaking my UI update after deletion.

**Comment by cpalmer-atx** [comment]
TLDR; my solution to a similar problem was to make sure the deleteGoal function in goalController.js on the back end was returning JUST the id of the goal to be deleted.  My controller was returning `Deleted id ${req.params.id}`, so the filter method in the deleted/fulfilled reducer couldn't weed out the goal to be deleted.

Detailed explanation:
Check your state in Redux dev tools when you delete a goal.  I had a similar problem where the goal wouldn't be deleted on the first delete attempt, but it would get deleted with a 400 error from the back end on the second attempt.  The first attempt would correctly call the delete/pending, delete/fulfilled reducers, so it lead me to believe there was something wrong with the way state.goals.filter() was updating state.  Long story short, the deleteGoal function in goalController was returning res.status(200).json({`Deleted id ${req.params.id}` }) instead of just the id, so the goal was being deleted from the database while not being removed from state on the front end.
